proxy-node-translator/src/main/java/uk/gov/ida/notification/translator/saml/HubResponseTranslator.java

```diff
@@ -87,6 +92,30 @@ public Response translate(HubResponseContainer hubResponseContainer) {
         );
     }
 
+    private void validateHubResponseContainerAttributes(HubResponseContainer hubResponseContainer) {
```

Comment:
https://github.com/alphagov/verify-proxy-node/blob/cef167e53539d5127dcade5e83d0109654de8afe/proxy-node-translator/src/main/java/uk/gov/ida/notification/translator/saml/HubResponseTranslator.java#L95-L117

This method is unnecessary.  `hubResponseContainer` itself will never be null because we create it ourself in the resource and then pass it on to the Translator. The `Attributes` object itself and all its properties have `@NotNull` flags in the DTOs so they will never be null either by definition.
The only check we need to do from this list is whether DoB has a value (it may be not null but still be missing value):
`hubResponseContainer.getAttributes().getDateOfBirth().getValue() == null`

Reply:
At the moment we don't know what the vsp response looks like for the other statuses as we've only seen the happy path.  In the stubbed unit test we can return whatever values we like, so I wanted the contract to be clear if anything is missing rather than just have an unexpected null pointer.

Reply:
The DTOs have `@NotNull` flags so we would never get an unexpected null pointer. In fact, we may need to remove some of those flags later we integrate with a real VSP as we discover when those attributes are null depending on the value of `scenario` attribute. Then we can add appropriate checks but at the moment we don't know what they should be and are not necessary IMO

Reply:
I built the validation when the need for it became obvious when writing the unit test.  It may be true when the code is integrated that certain scenarios will never happen, which is good.  But when unit testing it's not always obviously how the code will end up being used.  There are valid unit test scenarios where these can be null, so I think it's good to define what happens if they are null.  The validation will help when we do the integration because it will give you explicit information about what's going wrong, so it makes life easier.  I have raised cards to get clarification on what the data looks like for the other scenarios.  Going through the testing process and thinking about what can go wrong helps to clarify requirements.